apps/web/src/app/(default)/hjem/_components/echo-birthday.tsx

```diff
@@ -0,0 +1,71 @@
+"use client";
+
+import { useEffect, useState } from "react";
+
+import ConfettiForBDay from "./confetti";
+
+export default function EchoBirthdayBanner() {
+  const today = new Date();
+  const echoBirthday = new Date(today.getFullYear(), 10, 7);
+  const [timeDifference, setTimeDifference] = useState(0);
+  const afterBirthday = new Date(today.getFullYear(), 10, 8);
+
+  // Durations in milliseconds
+  const startDate = new Date(today.getFullYear(), 9, 7);
+  const progressDuration = echoBirthday.getTime() - startDate.getTime();
+  const progressElapsed = new Date().getTime() - startDate.getTime();
+
+  //calcualte percenrtage complete
+  let percenrtage = 0;
+  if (progressDuration > 0) {
+    percenrtage = (progressElapsed / progressDuration) * 100;
+  }
+
+  //progressbar completion style
+  const progressbar = (percenrtage = Math.max(0, Math.min(100, percenrtage)));
+
+  useEffect(() => {
+    const intervalId = setInterval(() => {
+      setTimeDifference(echoBirthday.getTime() - new Date().getTime());
+    }, 1000);
+    return () => clearInterval(intervalId);
+  }, []);
```

Comment:
Her klager linteren på at du mangler en funksjon i dependency arrayet ditt. Her kanskje faktisk bruke en helt ny react feature som kom for en uke siden for å fikse det.

```ts
const onTick = useEffectEvent(() => {
  setTimeDifference(echoBirthday.getTime() - new Date().getTime());
});

useEffect(() => {
  const intervalId = setInterval(() => {
    onTick();
  }, 1000);

  return () => clearInterval(intervalId);
}, []);
```

https://react.dev/learn/separating-events-from-effects#limitations-of-effect-events